src/foam/dao/index/TreeIndex.java

```diff
@@ -96,9 +96,9 @@ public Object bulkLoad(FObject[] a) {
       }
       // use partialEval to simplify predicate themselves.
       predicate = predicate.partialEval();
-      if ( predicate instanceof True ) return new Object[]{state, null};
     }
 
+    if ( predicate instanceof True ) predicate = null;
```

Comment:
I so hate this style of no {} with the if.  Had to go look at the entire source to understand where the final return went.  An empty line would help in future.